### PR TITLE
Document android multithreaded platform view constraints

### PR DESCRIPTION
**Pull request by reidbaker**
Update documentation based on the investigation done in 
https://b.corp.google.com/issues/299285351

This constraint is not listed in the places plugin authors were looking for it. 

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by reidbaker** [comment]
> Visit the preview URL for this PR (updated for commit [bdd13ef](https://github.com/flutter/website/commit/bdd13ef46dec5a94cdcd5411d7f39a8376780b13)):
> 
> https://flutter-docs-prod--pr9373-multithread-android-3si1hplo.web.app
> 
> (expires Mon, 18 Sep 2023 14:37:04 GMT)
> 
> 🔥 via [Firebase Hosting GitHub Action](https://github.com/marketplace/actions/deploy-to-firebase-hosting) 🌎
> 
> Sign: d5ba327eec813901cac8396c4f458b02288624ab

The specific page that was changed. 
https://flutter-docs-prod--pr9373-multithread-android-3si1hplo.web.app/platform-integration/android/platform-views

File: src/platform-integration/android/platform-views.md [hunk 1]
```diff
@@ -406,6 +406,10 @@ android {
 }
 ```
 
+#### Multi threaded draw calls
+
+Android's PlatformView implementation requires that the view is manually invalidated when the Android View draws using a separate thread.
```
Comment (atsansone) [inline src/platform-integration/android/platform-views.md:411]: **issue:** Needs semantic line breaks. sembr.org

Needs to follow confition, then action.

```markdown
When the Android View draws using a separate thread,
you must invalidate the view according to 
Android's PlatformView implementation. 
```

**Review by atsansone — COMMENTED** [review]
Recommended some changes.

**Review by atsansone — APPROVED** [review]

File: src/platform-integration/android/platform-views.md [hunk 2]
```diff
@@ -406,6 +406,12 @@ android {
 }
 ```
 
+#### Multi threaded draw calls
```
Comment (johnmccutchan) [inline src/platform-integration/android/platform-views.md:409]: s/Multi threaded/Multithreaded

**Comment by reidbaker** [comment]
Holding merge until @johnmccutchan has a chance to review.

File: src/platform-integration/android/platform-views.md [hunk 3]
```diff
@@ -406,6 +406,12 @@ android {
...
+
+When the Android View draws using a separate thread,
```
Comment (johnmccutchan) [inline src/platform-integration/android/platform-views.md:411]: This requirement doesn't come from having a background thread render the view but from the views themselves. I would rephrase:

Certain Android Views do not invalidate themselves when their content changes. Some example views include `SurfaceView` and `SurfaceTexture`. When your Platform View includes these views you are required to manually invalidate the view after they have been drawn to (or more specifically: after the swap chain is flipped). Manual view invalidation is done by calling `invalidate` on the View or one of its parent views.

**Comment by johnmccutchan** [comment]
Leaving a non-actionable comment here for posterity. 

1. HybridComposition mode does not require manual view invalidation but suffers from major performance problems among other missing features. 
2. VirtualDisplay backed Platform Views do not require manual view invalidation but suffers from a11y issues as well as other subtle compatibility issues (because VirtualDisplay backed Platform Views don't have access to / aren't embedded in the real view hierarchy of the application).

My plan with the graphics team is deprecate (1) and with manual view invalidation we don't need (2).

**Comment by reidbaker** [comment]
> Leaving a non-actionable comment here for posterity.
> 
> 1. HybridComposition mode does not require manual view invalidation but suffers from major performance problems among other missing features.
> 2. VirtualDisplay backed Platform Views do not require manual view invalidation but suffers from a11y issues as well as other subtle compatibility issues (because VirtualDisplay backed Platform Views don't have access to / aren't embedded in the real view hierarchy of the application).
> 
> My plan with the graphics team is deprecate (1) and with manual view invalidation we don't need (2).

I will also add that those are the only 2 modes defined on this page. Textured Layer Hybrid Composition is not even mentioned. Which I considered before updating and decided that it was better for someone looking for this information to have a hint that invalidation might help than to have no mention of it.

File: src/platform-integration/android/platform-views.md [hunk 4]
```diff
@@ -406,6 +406,16 @@ android {
 }
 ```
 
+#### Multithreaded draw calls
```
[the remark this answers is not in the document]
Comment (reidbaker) [reply src/platform-integration/android/platform-views.md]: done

File: src/platform-integration/android/platform-views.md (hunk 3 above)
Comment (reidbaker) [reply src/platform-integration/android/platform-views.md]: done

File: src/platform-integration/android/platform-views.md (hunk 2 above)
Comment (reidbaker) [reply src/platform-integration/android/platform-views.md]: change title
